.golangci.yml

```diff
@@ -0,0 +1,248 @@
+# using a custom configuration with all non-essential linters disabled,
+# because the default cfg is too strict and requires too much refactoring
+
+run:
+  go: '1.18'
+  concurrency: 4
+  timeout: 2m
+  issues-exit-code: 1
+
+  # default is true. Enables skipping of directories:
+  #   vendor$, third_party$, testdata$, examples$, Godeps$, builtin$
+  skip-dirs-use-default: true
+  skip-dirs:
+    - "test"
+  skip-files: [".*_test.go$"]
+  tests: true # whether to include test files
+
+
+output:
+  format: tab
+  print-issued-lines: true
+  print-linter-name: true
+  unique-by-line: true
+  sort-results: true
+
+
+linters-settings:
+  cyclop:
+    max-complexity: 20 # Default: 10
+    package-average: 0.5 # Default: 0.5
+    skip-tests: true
+
+  decorder:
+    dec-order:
+      - type
+      - const
+      - var
+      - func
+    disable-dec-order-check: false
+    disable-init-func-first-check: false
+    disable-dec-num-check: true # true allows multiple var/type/etc declarations
+
+  errcheck:
+    check-type-assertions: true
+    check-blank: true
+
+  errchkjson:
+    check-error-free-encoding: true
+    report-no-exported: true
+
+  exhaustruct:
```

Comment:
Does this replace https://github.com/jiping-s/exhaustivestruct or is there no struct requiring the latter in slog-agent? There were structures for fluentd forward protocol but they might have been completely replaced by manual serialization (instead of reflection-based approach)